### ISSUE TITLE
cosmos SDK v0.51.1 changes to NewInterfaceRegistry()

### ISSUE THREAD

**Issue by KyleMoser** [cosmos/relayer#1385]
There is a potential bug in the relayer due to calling `NewInterfaceRegistry()`. Calling `NewInterfaceRegistry()` provides a default address conversion codec which will throw an error any time the codec is used to convert addresses, such as when calling `tx.FeePayer()`.

Here is the new code for NewInterfaceRegistry, note the `failingAddressCodec` provided by SDK team:

```
registry, err := NewInterfaceRegistryWithOptions(InterfaceRegistryOptions{
		ProtoFiles: proto.HybridResolver,
		SigningOptions: signing.Options{
			AddressCodec:          failingAddressCodec{},
			ValidatorAddressCodec: failingAddressCodec{},
		},
	})
```

This should likely be e.g.

```
import xcodec "github.com/cosmos/cosmos-sdk/x/auth/codec"
interfaceRegistry, err := types.NewInterfaceRegistryWithOptions(types.InterfaceRegistryOptions{
		ProtoFiles: proto.HybridResolver,
		SigningOptions: signing.Options{
			AddressCodec:          xcodec.NewBech32Codec(accBech32Prefix),
			ValidatorAddressCodec: xcodec.NewBech32Codec(valBech32Prefix),
		},
	})
```

**Comment by akc2267** [comment]
@KyleMoser this is for sdk 0.50.1 right?

**Comment by KyleMoser** [comment]
> @KyleMoser this is for sdk 0.50.1 right?

Good catch. It goes back even further than that. 0.50.0 also has this issue. So I have no idea how far back it goes but it does affect 0.51.1 which rly is using. That said, this is now fixed by relayer release 2.5.0, so I'm closing this :)
